### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by fb55** [cheeriojs/cheerio#3985] [General]
Cheerio 1.0 is here! 🎉

# [Announcement Blog Post](https://cheerio.js.org/blog/2024-08-07-version-1/)

## Breaking Changes

- The minimum NodeJS version is now 18.17 or higher https://github.com/cheeriojs/cheerio/pull/3959
- Import paths were simplified. For example, use `cheerio/slim` instead of
  `cheerio/lib/slim`. https://github.com/cheeriojs/cheerio/pull/3970
- The deprecated default Cheerio instance and static methods were removed. https://github.com/cheeriojs/cheerio/pull/3974

  Before, it was possible to write code like this:

  ```ts
  import cheerio, { html } from 'cheerio';

  html(cheerio('<test></test>')); // ~ '<test></test>' -- NO LONGER WORKS
  ```

  Make sure to always load documents first:

  ```ts
  import * as cheerio from 'cheerio';

  cheerio.load('<test></test>').html();
  ```

- Node types previously re-exported by Cheerio must now be imported directly
  from (`domhandler`)(https://github.com/fb55/domhandler). https://github.com/cheeriojs/cheerio/pull/3969

- htmlparser2 options now reside exclusively under the `xml` key (https://github.com/cheeriojs/cheerio/pull/2916):

  ```ts
  const $ = cheerio.load('<html>', {
    xml: {
      withStartIndices: true,
    },
  });
  ```

## New Features

* Add functions to load buffers, streams & URLs in NodeJS by @fb55 in https://github.com/cheeriojs/cheerio/pull/2857
* Add extract method by @fb55 in https://github.com/cheeriojs/cheerio/pull/2750

## Fixes

* Allow imports of `cheerio/utils` by @blixt in https://github.com/cheeriojs/cheerio/pull/2601
* Allow empty string in `data`, and simplify by @fb55 in https://github.com/cheeriojs/cheerio/pull/2818
* Make `closest` be able to start from text nodes by @Qualtagh in https://github.com/cheeriojs/cheerio/pull/2811
* Fix potential github action smells by @ceddy4395 in https://github.com/cheeriojs/cheerio/pull/3826

## Other

* Cheerio has [a new website](https://cheerio.js.org), featuring updated API docs and guides! https://github.com/cheeriojs/cheerio/pull/2950

**Full Changelog**: https://github.com/cheeriojs/cheerio/compare/v1.0.0-rc.12...v1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/cheeriojs/cheerio/releases/tag/v1.0.0'>v1.0.0</a>.</em>

**Comment by michaelfaith** [comment]
Hell yeah!  Stoked to this land.

**Comment by sefinek** [comment]
AFTER 2 YEARS! THANKS ^_^

**Comment by Alevale** [comment]
This was a big mistake as seen by the issues in the package, please move this release to 2.0.0 and promote `1.0.0-RC12` as stable 1.0.0

The longer you keep things this way the more people are going to have an issue when the wrongly tagged and newly added functionality for 1.0.0 isn't available anymore after the revert.

@fb55

**Comment by gongshun** [comment]
my project auto installed cheerio version 1.0.0 because I did not lock the version, and then reported an error

<img width="1241" alt="image" src="https://github.com/user-attachments/assets/fb58c4c2-f27b-41da-b2da-4988a57eaa2c">

After checking, it was because I installed [@vue/server-test-utils](https://www.npmjs.com/package/@vue/server-test-utils?activeTab=code)，@vue/server-test-utils depends on cheerio version 1.0.0-rc.2 ，cheerio released 1.0.0 with breaking changes. 1.0.0 requires node18 but I am using node 16, which causes code errors.

<img width="693" alt="image" src="https://github.com/user-attachments/assets/38f9bd26-b5da-4cb0-824c-19e29de23474">

**Comment by laymonage** [comment]
The blog post link is outdated and returns 404. New link: https://cheerio.js.org/blog/2024-08-07-version-1/
